contracts/l1/interfaces/IStarknetMessaging.sol

```diff
@@ -12,7 +12,7 @@ interface IStarknetMessaging is IStarknetMessagingEvents {
         uint256 toAddress,
         uint256 selector,
         uint256[] calldata payload
-    ) external returns (bytes32);
+    ) external returns (bytes32, uint256);
```

Comment:
According to the [0.9.0 latest upgrade](https://github.com/starkware-libs/starkgate-contracts/commit/c08863a1f08226c09f1d0748124192e848d73db9), it should returns the `nonce` as well.

Reply:
reverted to using [l1tol2MessagesNonce](https://github.com/starkware-libs/starkgate-contracts/blob/c08863a1f08226c09f1d0748124192e848d73db9/src/starkware/starknet/solidity/StarknetMessaging.sol#L45) since the contract on devnet isn't yet updated and returns the msgHash only.

Reply:
But what about the deployed messaging contract in production? Is it upgraded to use the nonce?